[Tin Jingyao] iP

tin-jy opened the pull request:
### **DukePro**
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

DukePro frees your mind of having to remember things you need to do. It's,

text-based
easy to learn
~~FAST~~ *SUPER FAST* to use
All you need to do is,

1. download it from [here](https://nus-cs2103-ay2223s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the `main` method:

```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

tin-jy committed:
Add ability to read inputs and echo it as output

tin-jy committed:
Add add and list functionality

tin-jy committed:
Add functionality to mark and unmark tasks

tin-jy committed:
Add different types of tasks

There are 3 types of tasks, Todos, Events, and Deadline tasks

tin-jy committed:
Add test cases for text-ui testing

tin-jy committed:
Add error handling for incorrect user inputs

tin-jy committed:
Add functionality to remove tasks

edricyeo commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,208 @@
+import java.util.Locale;
+import java.util.Scanner;
+
 public class Duke {
```
Perhaps apply OOP to split the Duke class into different smaller classes?

domoberzin replied:
Could split the class into various classes, i.e. TaskList, Event, Deadline, instead of putting them into one class

domoberzin replied:
Could also add JavaDocs for documentation purposes

edricyeo commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,208 @@
+
+    private static final ArrayList<Task> LIST_OF_THINGS = new ArrayList<>();
+
+    private abstract static class Task {
```
Maybe you could create Task as a separate class, instead of nesting it in the Duke class.

edricyeo commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,208 @@
+            }
+        }
+
+        private static class EventTask extends Task {
```
I like how you named the different Tasks clearly! 👍

edricyeo commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,208 @@
+            return false;
+        }
+        try {
+            Integer.parseInt(str.substring(7));
```
Perhaps you could consider using str.split(" ") rather than str.substring(7). May improve reusability of code instead of having to change the number according to the length of the command?

edricyeo commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,208 @@
+
 public class Duke {
+
+    private static final ArrayList<Task> LIST_OF_THINGS = new ArrayList<>();
```
Maybe you could consider extracting this out as a separate TaskList class, which would handle adding tasks, etc.

edricyeo reviewed (commented):
Generally good effort!
I like the clear naming of classes, variables and methods.
I think generally you could split your code into different classes, and include JavaDocs for your classes and methods!

domoberzin commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,208 @@
+    private static final ArrayList<Task> LIST_OF_THINGS = new ArrayList<>();
+
+    private abstract static class Task {
+        private final String task_name;
```
Should be camelcase formatting for variables

domoberzin reviewed (commented):
Coding standard is generally adhered to w.r.t naming, but could extract out the different classes, as well as add JavaDocs. :)

tin-jy committed:
Add a FileHandler class to deal with saving inputs in a local file

Shifted inner class Task to a new file

tin-jy committed:
Add functionality to save task list into a local file on every change

tin-jy committed:
Add functionality for Duke to load task list on startup

tin-jy committed:
Add ability for Duke to verify dates and time

tin-jy committed:
Add classes Parser, Ui, TaskList to include more OOP

tin-jy committed:
Place all Duke related Java files into packages

DarrenCsAcc commented on src/main/java/duke/Command.java:
```diff
@@ -0,0 +1,5 @@
+package duke;
+
+public class Command {
+
+}
```
I am curious why is this empty?

tin-jy committed:
Merge branch 'add-gradle-support' of https://github.com/nus-cs2103-AY2223S1/ip

tin-jy committed:
Add more packages and classes for organisation

tin-jy committed:
Add more classes and error handling

tin-jy committed:
Add JUnit tests for public methods in some classes

tin-jy committed:
Update dependencies in build.gradle

tin-jy committed:
Add JavaDocs

tin-jy committed:
Correct style issues to adhere to coding standards

tin-jy committed:
Add ability to search for tasks using keywords

tin-jy committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/duke/Duke.java
#	src/main/java/duke/task/TodoTask.java

tin-jy committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/duke/Parser.java

tin-jy committed:
Add checkstyle.xml and suppressions.xml for checkstyle

tin-jy committed:
Fix a bug with the find command

tin-jy committed:
Add test cases

tin-jy committed:
Create JAR file

tin-jy committed:
Add a basic GUI for Duke

tin-jy committed:
Fully implement GUI and allow Duke to respond to commands

tin-jy committed:
Merge branch 'branch-A-CheckStyle'

# Conflicts:
#	text-ui-test/EXPECTED.TXT
#	text-ui-test/input.txt

tin-jy committed:
Add package for GUI related java files

tin-jy committed:
Add more JUnit tests

tin-jy committed:
Add assert statements

tin-jy committed:
Improve code quality

Let's shorten excessively long methods by abstracting out certain parts
into separate methods.

tin-jy committed:
Merge pull request #1 from tin-jy/branch-A-Assertions

Add assert statements

tin-jy committed:
Merge branch 'master' into branch-A-CodeQuality

# Conflicts:
#	src/main/java/duke/Parser.java

tin-jy committed:
Merge pull request #2 from tin-jy/branch-A-CodeQuality

Branch a-code-quality

tin-jy committed:
Add functionality to change the file path for local storage

Flexibility to change the local file path through Duke allows users to
set new file locations without restarting Duke. Users may also create
new files this way.

tin-jy committed:
Change some output messages

tin-jy committed:
Add some bugfixes

Add some JavaDocs

tin-jy committed:
Fix checkstyle issues

tin-jy committed:
Change profile images for Duke and the user.

tin-jy committed:
Add a user guide in README.md

tin-jy committed:
Fix bug related to local file path
